src/services/earn/types.ts

```diff
@@ -0,0 +1,28 @@
+import { PermitData } from '@services/permit2';
+import { Address, BigIntish, ChainId, TimeString, TokenAddress } from '@types';
+import { Hex } from 'viem';
+
+export type IEarnService = {};
+
+export type CreateEarnPositionParams = {
+  chainId: ChainId;
+  strategyId: bigint;
+  depositToken: TokenAddress;
+  owner: Address;
+  permissions: EarnPermissionSet[];
+  strategyValidationData: Hex;
+  misc: Hex;
```

Comment:
Let's make these two optional, and default to an empty string when they are not set